service/src/test/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentServiceTest.java

```diff
@@ -0,0 +1,532 @@
+    }
+
+    @Test
+    void shouldReturnFurtherEvidenceCollectionWhenFurtherEvidenceIsNotRelatedToHearingAndCollectionIsPresent() {
+        List<Element<SupportingEvidenceBundle>> furtherEvidenceBundle = buildSupportingEvidenceBundle();
+
+        CaseData caseData = CaseData.builder()
+            .manageDocument(buildManagementDocument(FURTHER_EVIDENCE_DOCUMENTS, NO.getValue()))
+            .furtherEvidenceDocuments(furtherEvidenceBundle)
+            .build();
+
+        List<Element<SupportingEvidenceBundle>> furtherDocumentBundleCollection =
+            manageDocumentService.getFurtherEvidenceCollection(caseData);
+
+        assertThat(furtherDocumentBundleCollection).isEqualTo(furtherEvidenceBundle);
+    }
+
+    @Test
+    void shouldReturnHearingEvidenceCollectionWhenFurtherEvidenceIsRelatedToHearingWithExistingEntryInCollection() {
+        List<Element<SupportingEvidenceBundle>> furtherEvidenceBundle = buildSupportingEvidenceBundle();
+        UUID hearingId = UUID.randomUUID();
+        List<Element<HearingBooking>> hearingBookings = List.of(element(hearingId, buildFinalHearingBooking()));
+
+        CaseData caseData = CaseData.builder()
+            .hearingDetails(hearingBookings)
+            .manageDocumentsHearingList(asDynamicList(hearingBookings, hearingId, hearing -> hearing.toLabel(DATE)))
+            .hearingFurtherEvidenceDocuments(List.of(
+                element(hearingId, HearingFurtherEvidenceBundle.builder()
+                    .supportingEvidenceBundle(furtherEvidenceBundle)
+                    .build()
+                )
+            ))
+            .manageDocument(buildManagementDocument(FURTHER_EVIDENCE_DOCUMENTS, YES.getValue()))
+            .build();
+
+        List<Element<SupportingEvidenceBundle>> furtherDocumentBundleCollection =
+            manageDocumentService.getFurtherEvidenceCollection(caseData);
+
+        assertThat(furtherDocumentBundleCollection).isEqualTo(furtherEvidenceBundle);
+    }
+
+    @Test
+    void shouldSetDateTimeUploadedOnNewCorrespondenceSupportingEvidenceItems() {
+        LocalDateTime yesterday = time.now().minusDays(1);
+        List<Element<SupportingEvidenceBundle>> correspondingDocuments = buildSupportingEvidenceBundle();
+        correspondingDocuments.add(element(SupportingEvidenceBundle.builder()
+            .dateTimeUploaded(yesterday)
+            .build()));
+
+        List<Element<SupportingEvidenceBundle>> updatedCorrespondingDocuments
+            = manageDocumentService.setDateTimeUploadedOnSupportingEvidence(correspondingDocuments, List.of());
+
+        List<SupportingEvidenceBundle> supportingEvidenceBundle = unwrapElements(updatedCorrespondingDocuments);
+
+        assertThat(supportingEvidenceBundle.get(0).getDateTimeUploaded()).isEqualTo(time.now());
+        assertThat(supportingEvidenceBundle.get(1).getDateTimeUploaded()).isEqualTo(yesterday);
+    }
+
+    @Test
+    void shouldSetNewDateTimeUploadedOnOverwriteOfPreviousDocumentUpload() {
```

Comment:
Should we add scenario, which ensure the overwrite doesn't happen if the file is not modified? I guess that should be functionality.

Reply:
Added